crates/http/src/routes/settings.rs

```diff
@@ -0,0 +1,86 @@
+use axum::{
+    extract::Query,
+    routing::{get, post},
+    Json, Router,
+};
+use iron_settings::{DarkMode, SerializedSettings};
+use iron_types::ChecksummedAddress;
+use serde::Deserialize;
+
+use crate::{Ctx, Result};
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SetSettings {
+    new_settings: SerializedSettings,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct DarkModeParams {
+    mode: DarkMode,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct FastModeParams {
+    mode: bool,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct GetAliasParams {
+    address: ChecksummedAddress,
+}
```

Comment:
I'm wondering. is there no way to avoid having these individual structs when it's just 1 or 2 fields?

feels extremely verbose